frontend/src/main/web/lib/components/glossary/DataTable.jsx

```diff
@@ -0,0 +1,438 @@
+  },
+
+  _renderTransCountHeader: function (label) {
+    var key = this.ENTRY.TRANS_COUNT.sort_field, asc = this._getSort(key);
+    return this._renderHeader(label, key, asc, true);
+  },
+
+  _renderHeader: function (label, key, asc, allowSort) {
+    return (<ColumnHeader value={label}
+      field={key}
+      key={key}
+      allowSort={allowSort}
+      sort={asc}
+      onClickCallback={this._onHeaderClick}/>);
+  },
+
+  _onHeaderClick: function (field, ascending) {
+    Actions.updateSortOrder(field, ascending);
+  },
+
+  _renderCell: function (resId, rowIndex, field, readOnly, placeholder) {
+    var key = this._generateKey(field.col, rowIndex, resId);
+    if (resId === null) {
+      return (<LoadingCell key={key}/>)
+    } else {
+      var entry = this._getGlossaryEntry(resId);
+      var value = _.get(entry, field.field);
+
+      if (readOnly) {
+        return (<span className="mh1/2" key={key}>{value}</span>)
+      } else {
+        return (<InputCell
+          value={value}
+          resId={resId}
+          key={key}
+          placeholder={placeholder}
+          rowIndex={rowIndex}
+          field={field.field}
+          onBlurCallback={this._onRowBlur}
+          onFocusCallback={this._onRowClick}/>);
+      }
+    }
+  },
+
+  _renderSourceCell: function (resId, cellDataKey, rowData, rowIndex,
+                               columnData, width) {
+    return this._renderCell(resId, rowIndex, this.ENTRY.SRC, true, '');
+  },
+
+  _renderTransCell: function(resId, cellDataKey, rowData, rowIndex,
+                             columnData, width) {
+    var readOnly = !this.props.canUpdateEntry, placeholder = 'enter a translation';
+    return this._renderCell(resId, rowIndex, this.ENTRY.TRANS, readOnly, placeholder);
+  },
+
+  _renderPosCell: function (resId, cellDataKey, rowData, rowIndex,
+                            columnData, width) {
+    var readOnly = !this.props.canUpdateEntry || this._isTranslationSelected(),
+      placeholder = 'enter part of speech';
+    return this._renderCell(resId, rowIndex, this.ENTRY.POS, readOnly, placeholder);
+  },
+
+  _renderDescCell: function (resId, cellDataKey, rowData, rowIndex,
+                             columnData, width) {
+    var readOnly = !this.props.canUpdateEntry || this._isTranslationSelected(),
+      placeholder = 'enter description';
+    return this._renderCell(resId, rowIndex, this.ENTRY.DESC, readOnly, placeholder);
+  },
+
+  _renderTransCountCell: function (resId, cellDataKey, rowData, rowIndex,
+                              columnData, width) {
+    return this._renderCell(resId, rowIndex, this.ENTRY.TRANS_COUNT, true, '');
+  },
+
+  _renderActionCell: function (resId, cellDataKey, rowData, rowIndex,
+                            columnData, width) {
+    var self = this;
+    if(resId === null) {
+      return (<LoadingCell/>);
+    } else if(!self.props.canUpdateEntry && !self.props.canAddNewEntry) {
+      return null;
```

Comment:
Can just use `return` with nothing if you want `undefined`.

When I see `return null` I wonder whether the code that calls this will do something special with exactly the value `null` and would behave wrong with `undefined` - I hope we don't have any code like that.